### PR TITLE
feat(#40): inbound requests page — add frontend unit tests

### LINKED ISSUE #40: Inbound Requests Page ("Couples That Want to Connect With You")
**Description:**
As a user, I want to see all couples who have sent us a connection request so my partner and I can each accept or decline.
 
**Acceptance Criteria:**
- [ ] `GET /connections/inbound` returns all requests at `INTEREST_ALIGNED` or `REQUEST_PENDING` status directed at my couple
- [ ] Each card shows the requesting couple's profile (names, `about_us`, location, tags), shared tag count, and my partner's response state (accepted / pending / declined)
- [ ] `PATCH /connections/:id/respond` works identically to #36 — accept transitions to `CONNECTED` when both partners in couple 2 accept; decline sets `DECLINED` and silently removes from both inbound pages
- [ ] Which partner declined is never disclosed to couple 1
- [ ] `CONNECTED` transitions move the entry out of this page and into the Connections view
- [ ] Unit tests cover: only `INTEREST_ALIGNED`/`REQUEST_PENDING` requests shown, both-accept triggers `CONNECTED`, single-decline triggers `DECLINED` silently, declined status not attributed to specific partner in couple 1 notification

### PR DESCRIPTION
**Pull request by edwardchanneu**
Closes #40

## Summary

- Adds `client/src/pages/InboundRequestsPage.test.tsx` with 11 unit tests covering all AC scenarios for the inbound requests page
- All backend logic (`GET /connections/inbound`, `POST /connections/:id/respond`) and the `InboundRequestsPage` component were already implemented on this branch; this PR closes the test gap

## Tests added

| Test | AC coverage |
|------|------------|
| Renders page heading | — |
| Placeholder when no requests | — |
| Renders card for each result | INTEREST_ALIGNED/REQUEST_PENDING shown |
| Accept + Decline buttons (my_response null) | — |
| "Waiting for partner" (my_response true) | — |
| Accept calls respond with `accept: true` | — |
| Decline calls respond with `accept: false` | — |
| "Connected!" after CONNECTED response | Both-accept triggers CONNECTED |
| "Declined" after decline | Single-decline triggers DECLINED silently |
| Error message on API failure | — |
| No partner attribution after decline | Declined status not attributed to specific partner |

## AI Disclosure

| Field | Value |
|-------|-------|
| % AI-generated | ~90% (Writer agent wrote the test file; lint fixes applied by human) |
| Tool used | Claude Code — claude-sonnet-4-6 (Anthropic) |
| Human review applied | Yes — user reviews before merge (Step 4 of Writer/Reviewer pattern) |

## C.L.E.A.R. Review

A Reviewer agent applied the C.L.E.A.R. framework to this diff. Findings are posted as inline review comments on this PR.

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: client/src/pages/InboundRequestsPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    });
+  });
+}
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+describe("InboundRequestsPage", () => {
+  it("renders page heading", async () => {
```
Comment (edwardchanneu) [inline client/src/pages/InboundRequestsPage.test.tsx:59]: **[E — Evidence] ⚠️ WARN: Heading test passes trivially without awaiting data load**

This test passes even if the component crashes during its async fetch, because the heading renders synchronously before any data arrives. It is not a meaningful regression guard.

**Fix:** Fold this assertion into the `renders a card for each result` test (assert heading present after `waitFor`), or add an `await waitFor` here so the test only passes once the page has fully loaded.

File: client/src/pages/InboundRequestsPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+        <InboundRequestsPage />
+      </MemoryRouter>,
+    );
+    await waitFor(() =>
+      expect(screen.getByRole("button", { name: /accept/i })).toBeInTheDocument(),
+    );
+    expect(screen.getByRole("button", { name: /decline/i })).toBeInTheDocument();
+  });
+
+  describe("polling tests (fake timers)", () => {
```
Comment (edwardchanneu) [inline client/src/pages/InboundRequestsPage.test.tsx:123]: **[L — Logic] ⚠️ WARN: Fake timers declared but polling never exercised**

This `describe` block enables fake timers, but no test inside advances the clock to fire the `setInterval` (component line 123: `setInterval(() => void poll(), 3000)`). The polling path — where `my_response: true` auto-refreshes to `CONNECTED` — is completely untested.

**Fix:** Add a test that renders with `my_response: true`, stubs the polled call to return the request absent, calls `await vi.advanceTimersByTimeAsync(3000)`, and asserts the card disappears.

File: client/src/pages/InboundRequestsPage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    );
+
+    await waitFor(() =>
+      expect(
+        screen.getByText(/failed to load inbound requests/i),
+      ).toBeInTheDocument(),
+    );
+  });
+
+  it("privacy: declined entry shows no partner attribution", async () => {
```
Comment (edwardchanneu) [inline client/src/pages/InboundRequestsPage.test.tsx]: **[A — Architecture] ⚠️ Note: ad-hoc timer setup outside the scoped `describe`**

`vi.useFakeTimers()` is called manually at line 353 and `vi.useRealTimers()` at line 399. If this test throws before line 399, all subsequent tests run with fake timers active.

**Fix:** Move this test inside the existing `describe("polling tests (fake timers)")` block to use its shared `beforeEach`/`afterEach`.

---

**[R — Risk] ⚠️ WARN: FR-CONN-08 privacy assertion is too narrow**

Checking for `"sam declined"` / `"riley declined"` passes even if the UI showed `"A partner declined"` or revealed the declining partner another way. Also, FR-CONN-08 is about not revealing the decliner to the *requesting* couple (couple 1), not to the couple who made the decision — the inbound couple already knows their own partner's choice.

**Fix:** Add a comment clarifying what is verified (generic label, no name attribution). Real FR-CONN-08 coverage belongs on the outbound/sent-requests page.

**Review by edwardchanneu — COMMENTED** [review]
## C.L.E.A.R. Review — `InboundRequestsPage.test.tsx`

> Reviewer: Claude Code (claude-sonnet-4-6) · Pattern: Writer/Reviewer (W12)

| Dimension | Finding |
|-----------|--------|
| **C**ontext | ✅ PASS |
| **L**ogic | ⚠️ WARN |
| **E**vidence | ⚠️ WARN |
| **A**rchitecture | ✅ PASS (one note) |
| **R**isk | ⚠️ WARN |

### Overall verdict: REQUEST_CHANGES (would block merge if from an external reviewer)

Three warn items should be addressed before merge (see inline comments). None cause false positives today, but they leave real regression gaps open.

File: client/src/pages/InboundRequestsPage.test.tsx [hunk 4]
```diff
@@ -74,327 +68,334 @@ describe("InboundRequestsPage", () => {
...
-      ).toBeInTheDocument(),
+      expect(screen.getByRole("button", { name: /accept/i })).toBeInTheDocument(),
     );
+    expect(screen.getByRole("button", { name: /decline/i })).toBeInTheDocument();
   });
 
-  describe("CoupleDetailModal integration", () => {
-    it("modal is not visible initially", async () => {
-      mockSuccess();
+  describe("polling tests (fake timers)", () => {
```
Comment (edwardchanneu) [inline client/src/pages/InboundRequestsPage.test.tsx:123]: **[L-1] ⚠️ CARRIED: Polling interval never fired**

This describe block uses `vi.useFakeTimers` but no test calls `await vi.advanceTimersByTimeAsync(3000)` to actually fire the `setInterval`. The polling path remains untested.

**Fix:** Add a test that seeds `my_response: true`, stubs the polled `/connections/inbound` to return an empty array, advances timers by 3000ms, and asserts the card disappears.

---

**[C-1] ⚠️ NEW (merge): Missing `CoupleDetailModal integration` describe block**

The merge added `CoupleDetailModal` to this page. `DiscoveryPage.test.tsx` has a dedicated `describe` block for modal integration (open on card click, close via × and overlay, content rendering, CTA behavior). This file has none of it, breaking pattern consistency.

**Minimum required tests to add:**
1. Clicking a `CoupleCard` opens the modal (card becomes visible)
2. Clicking the close button (×) hides the modal
3. `hideCta` is respected — no "I'm interested" button in the modal on this page

**Review by edwardchanneu — COMMENTED** [review]
## C.L.E.A.R. Review (Post-Merge) — `InboundRequestsPage.test.tsx`

> Reviewer: Claude Code (claude-sonnet-4-6) · Writer/Reviewer Pattern (W12) · Re-review after merge of `develop`

### Merge correctness

The merge is **clean** — no conflicts, TypeScript compiles, ESLint clean, all 11 tests pass. However, the merge introduced `CoupleDetailModal` into the component (clicking a card now opens a detail modal), and **zero new tests cover this new behavior**.

### Findings summary

| ID | Dim | Status | Description |
|----|-----|--------|-------------|
| L-1 | Logic | Carried | Polling interval never advanced in fake-timer tests |
| L-2 | Logic | Carried | No test for double-submit guard on `handleRespond` |
| L-3 | Logic | Carried | No test for `partner1: null` fallback rendering |
| E-1 | Evidence | Carried | Heading test passes trivially without async |
| A-1 | Arch | Carried | Privacy test manages fake timers ad-hoc outside scoped describe |
| R-1 | Risk | Carried | FR-CONN-08 privacy test checks wrong direction/actor |
| C-1 | Context | **NEW** | No `CoupleDetailModal integration` describe block — diverges from `DiscoveryPage` baseline |
| L-5 | Logic | **NEW** | No test: clicking CoupleCard opens modal |
| L-6 | Logic | **NEW** | No test: closing modal clears `selectedId` |
| L-7 | Logic | **NEW** | No test: `hideCta` prop enforced (CTA absent in inbound modal) |
| E-2 | Evidence | **NEW** | Entire modal path has zero coverage |
| R-2 | Risk | **NEW** | `isInterested={false}` hardcoded in modal props — latent bug if `hideCta` ever removed |

### Overall verdict: REQUEST_CHANGES

Minimum required before approval: modal open/close/hideCta tests + `partner1: null` fallback. See inline comments for specifics.

File: client/src/pages/InboundRequestsPage.test.tsx [hunk 5]
```diff
@@ -62,8 +56,8 @@ beforeEach(() => {
 });
 
 describe("InboundRequestsPage", () => {
-  it("renders the page heading", async () => {
-    mockSuccess();
+  it("renders page heading", async () => {
```
Comment (edwardchanneu) [inline client/src/pages/InboundRequestsPage.test.tsx:59]: **[E-1] ⚠️ CARRIED: Heading test passes trivially**

The heading renders synchronously before the async fetch — this test passes even if the component crashes during data loading. Not a meaningful regression guard.

**Fix:** Assert the heading inside the `renders a card for each result` test after `waitFor`, or add an `await waitFor(...)` here.

File: client/src/pages/InboundRequestsPage.test.tsx [hunk 6]
```diff
@@ -74,327 +68,334 @@ describe("InboundRequestsPage", () => {
...
+  });
 
-    it("closes the modal when the overlay backdrop is clicked", async () => {
-      mockSuccess();
-      render(
-        <MemoryRouter>
-          <InboundRequestsPage />
-        </MemoryRouter>,
-      );
+  it("privacy: declined entry shows no partner attribution", async () => {
```
Comment (edwardchanneu) [inline client/src/pages/InboundRequestsPage.test.tsx]: **[A-1] ⚠️ CARRIED: Ad-hoc fake timers outside scoped describe — fragile teardown**

`vi.useFakeTimers()` is called at line 353 and `vi.useRealTimers()` at line 399. If the test throws before line 399, all subsequent tests run with fake timers active.

**Fix:** Move this test inside the `describe("polling tests (fake timers)")` block above to use the shared `beforeEach`/`afterEach`.

---

**[R-1] ⚠️ CARRIED: FR-CONN-08 privacy test checks wrong actor**

FR-CONN-08 requires that the *requesting couple* (couple 1) cannot see which specific partner on couple 2 declined. This test verifies the *responding user's own* UI — a direction that is not the privacy concern FR-CONN-08 describes.

**Fix:** Add a comment clarifying the test's actual scope (generic "Declined" label, no name attribution on the responding user's view). Real FR-CONN-08 coverage belongs on the outbound/sent-requests page.

---

**[R-2] ⚠️ NEW (merge): `isInterested={false}` hardcoded in modal props**

`InboundRequestsPage.tsx` line ~249 passes `isInterested={false}` to the modal always, relying on `hideCta` to suppress the button. If `hideCta` is ever accidentally removed, the CTA would render in the wrong state for requests where the user has already accepted. A `hideCta` enforcement test (see C-1) would catch this regression.

**Comment by edwardchanneu** [comment]
## Writer Agent: Review findings addressed

All 12 C.L.E.A.R. findings from the post-merge review have been resolved in commit `79f05fe`. Test count: **11 → 18 tests**, all passing.

| Finding | Fix |
|---------|-----|
| E-1: Heading test trivially passes | Added `await waitFor` so test only passes after async load |
| L-1: Polling interval never fired | Added test using `vi.advanceTimersByTimeAsync(3100)` that asserts card disappears after poll returns `[]` |
| L-2: Double-submit guard untested | Added test seeding `my_response: true`, asserting respond endpoint never called |
| L-3: `partner1: null` fallback untested | Added test with `partner1: null` fixture, asserts `"? & Riley"` renders without crash |
| A-1: Ad-hoc fake timers in privacy test | Moved privacy test inside `describe("polling tests (fake timers)")` to use shared `beforeEach`/`afterEach` |
| R-1: FR-CONN-08 wrong actor | Added comment clarifying scope; FR-CONN-08 backend coverage noted |
| C-1/L-5/L-6/L-7/E-2: Zero modal coverage | Added `describe("CoupleDetailModal integration")` with 4 tests: open on card click, close via ×, close via overlay, `hideCta` enforced |

**Comment by edwardchanneu** [comment]
## Writer Agent: Review findings addressed

All 12 C.L.E.A.R. findings from the post-merge review have been resolved in commit 79f05fe. Test count: **11 → 18 tests**, all passing.

| Finding | Fix |
|---------|-----|
| E-1 | Added await waitFor so heading test only passes after async load |
| L-1 | Added polling test using vi.advanceTimersByTimeAsync(3100); asserts card disappears |
| L-2 | Added guard test seeding my_response:true; asserts respond endpoint never called |
| L-3 | Added partner1:null fixture test; asserts '? & Riley' renders without crash |
| A-1 | Moved privacy test inside polling describe block to use shared beforeEach/afterEach |
| R-1 | Added comment clarifying FR-CONN-08 scope; backend coverage noted |
| C-1/L-5/L-6/L-7/E-2 | Added CoupleDetailModal integration describe: open on card click, close via x, close via overlay, hideCta enforced |

**Review by edwardchanneu — COMMENTED** [review]
## C.L.E.A.R. Review (Revision Pass) — `InboundRequestsPage.test.tsx`

> Reviewer: Claude Code (claude-sonnet-4-6) · Writer/Reviewer Pattern (W12) · Verifying Writer's fixes

### Previous findings — all resolved ✅

| ID | Finding | Status |
|----|---------|--------|
| E-1 | Heading test trivially passes | ✅ RESOLVED |
| L-1 | Polling interval never fired | ✅ RESOLVED |
| L-2 | Double-submit guard untested | ✅ RESOLVED |
| L-3 | `partner1: null` fallback untested | ✅ RESOLVED |
| A-1 | Ad-hoc fake timers in privacy test | ✅ RESOLVED |
| R-1 | FR-CONN-08 wrong actor | ✅ RESOLVED |
| C-1/L-5/L-6/L-7/E-2 | Zero modal coverage | ✅ RESOLVED — 4 tests added |

The Writer made real, substantive improvements. All seven prior findings are correctly addressed.

**Review by hcarminati — APPROVED** [review]
